spring-auto-restdocs-core/src/test/java/capital/scalable/restdocs/request/RequestHeaderSnippetTest.java

```diff
@@ -0,0 +1,168 @@
+                .attribute(HandlerMethod.class.getName(), handlerMethod)
+                .build());
+    }
+
+    @Test
+    public void noHandlerMethod() throws Exception {
+        this.snippets.expect(REQUEST_HEADERS).withContents(equalTo("No headers."));
+
+        new RequestHeaderSnippet().document(operationBuilder
+                .build());
+    }
+
+    @Test
+    public void failOnUndocumentedHeaders() throws Exception {
+        HandlerMethod handlerMethod = createHandlerMethod("updateItem", Integer.class, String.class,
+                int.class);
+        initParameters(handlerMethod);
+
+        thrown.expect(SnippetException.class);
+        thrown.expectMessage(
+                "Following request headers were not documented: [id, subid, partId]");
+
+        new RequestHeaderSnippet().failOnUndocumentedParams(true).document(operationBuilder
+                .attribute(HandlerMethod.class.getName(), handlerMethod)
+                .attribute(JavadocReader.class.getName(), javadocReader)
+                .attribute(ConstraintReader.class.getName(), constraintReader)
+                .build());
+    }
+
+    @Test
+    public void deprecated() throws Exception {
+        HandlerMethod handlerMethod = createHandlerMethod("removeItem", int.class);
+        initParameters(handlerMethod);
+        mockParamComment("removeItem", "index", "item's index");
+
+        this.snippets.expect(REQUEST_HEADERS).withContents(
+                tableWithHeader("Parameter", "Type", "Optional", "Description")
+                        .row("index", "Integer", "false", "**Deprecated.**\n\nItem's index."));
+
+        new RequestHeaderSnippet().document(operationBuilder
+                .attribute(HandlerMethod.class.getName(), handlerMethod)
+                .attribute(JavadocReader.class.getName(), javadocReader)
+                .attribute(ConstraintReader.class.getName(), constraintReader)
+                .build());
+    }
+
+    private void initParameters(HandlerMethod handlerMethod) {
+        for (MethodParameter parameter : handlerMethod.getMethodParameters()) {
+            parameter.initParameterNameDiscovery(new DefaultParameterNameDiscoverer());
+        }
+    }
+
+    private void mockParamComment(String methodName, String parameterName, String comment) {
+        when(javadocReader.resolveMethodParameterComment(TestResource.class, methodName,
+                parameterName)).thenReturn(comment);
+    }
+
+    private HandlerMethod createHandlerMethod(String name, Class<?>... parameterTypes)
+            throws NoSuchMethodException {
+        return new HandlerMethod(new TestResource(), name, parameterTypes);
+    }
+
+    private static class TestResource {
+
+        @RequestMapping(value = "/items/{id}/subitem/{subid}/{partId}")
+        public void updateItem(@RequestHeader Integer id,
+                @RequestHeader("subid") String otherId,
+                @RequestHeader(required = false) int partId) { // required anyway
```

Comment:
The comment could be improved to `required anyway, because it's a primitive type`.

It would be good if you could also add a test for the other case where required is actually false, e.g.
`@RequestHeader(required = false) String yetAnotherId`